src/main/java/seedu/address/logic/commands/suggestions/AddNusModCommandSuggester.java

```diff
@@ -39,6 +43,30 @@
                     .filter(moduleCode -> {
                         return moduleCode.startsWith(searchModuleCode);
                     }).collect(Collectors.toUnmodifiableList());
+        } else if (prefix.equals(CliSyntax.PREFIX_LESSON_TYPE_AND_NUM)) {
+            final Optional<Module> optionalSelectedModule = getSelectedModule(model, arguments);
+            if (optionalSelectedModule.isEmpty()) {
+                return null;
+            }
+
+            final String userInput = commandArgument.getValue();
+
+            final int classSubsectionStartIdx = userInput.lastIndexOf(",") + 1;
+            final String suggestingSubsection = userInput.substring(classSubsectionStartIdx);
+            final String everythingPrior = userInput.substring(0, classSubsectionStartIdx);
+
+            final Module selectedModule = optionalSelectedModule.get();
+            final SemesterNo semesterNo = model.getSemesterNo();
+            final List<Lesson> moduleTimetable = selectedModule.getSemester(semesterNo).getTimetable();
+            return moduleTimetable
+                    .stream()
+                    .map(Lesson::getLessonTypeAndNoString)
+                    .distinct() // not sure why this is needed
```

Comment:
Was there an issue if you exclude `.distinct()`?

Reply:
Without the `.distinct()`, the `addmod m/CS2103T cl/LEC:G02` command continued to suggest four separate `LEC:G02` values. I'll continue debugging but for now, are you able to replicate the same issue?